core/envelope-sim/src/test/java/fr/sncf/osrd/envelope/ConstraintBuilderTest.java

```diff
@@ -78,6 +77,31 @@ void testSpeedCeilingConstraint() {
         assertEquals(1, builder.lastIntersection);
     }
 
+    @Test
+    void testSpeedMaintainConstraint() {
+        var partBuilder = new EnvelopePartBuilder();
+        var constraint = new SpeedConstraint(2, MAINTAIN_SPEED);
+        var builder = new ConstrainedEnvelopePartBuilder(
+                partBuilder,
+                constraint
+        );
+        assertTrue(constraint.initCheck(1, 2, 2));
+        assertTrue(builder.initEnvelopePart(2, 2, 1));
+        assertTrue(builder.addStep(2.5, 2));
+        assertFalse(builder.addStep(3, 1));
+        assertEquals(0, builder.lastIntersection);
```

Comment:
We test that it stops if we go below the limit, but we don't test what happens if it goes above the limit

Reply:
I'm adding a line